File: Examples/test-suite/octave/li_std_containers_overload_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

li_std_containers_overload

function check(got, expected)
  if (!strcmp(got, expected))
    error("Failed check. '%s' != '%s'", got, expected)
  endif
end

v = VectorX();
check(VectorOverload(v), "vector<X>");

v = VectorY();
check(VectorOverload(v), "vector<Y>");

v = VectorInt();
check(VectorOverload(v), "vector<int>");

v = VectorString();
check(VectorOverload(v), "vector<string>");

# TODO: Conversion from an Octave sequence not implemented yet
# v = {X()};
# check(VectorOverload(v), "vector<X>");

# v = {Y()};
# check(VectorOverload(v), "vector<Y>");

# v = {1, 2, 3};
# check(VectorOverload(v), "vector<int>");

# v = {"aaa", "bbb", "ccc"};
# check(VectorOverload(v), "vector<string>");

File: Examples/test-suite/lua/li_std_containers_overload_runme.lua
local v=require("li_std_containers_overload")

function check(got, expected)
  assert(got == expected, "Failed check. " .. got .. " != " .. expected)
end

local vec = v.VectorX()
check(v.VectorOverload(vec), "vector<X>")

vec = v.VectorY()
check(v.VectorOverload(vec), "vector<Y>")

vec = v.VectorInt()
check(v.VectorOverload(vec), "vector<int>")

vec = v.VectorString()
check(v.VectorOverload(vec), "vector<string>")

--[[ TODO Lua do not translate Lua table to vector<>
vec = { v.X() }
check(v.VectorOverload(vec), "vector<X>")
vec = { Y() }
check(v.VectorOverload(vec), "vector<Y>")
vec = { 1, 2, 3 }
check(v.VectorOverload(vec), "vector<int>")
vec = { "aaa", "bbb", "ccc" }
check(v.VectorOverload(vec), "vector<string>")
]]
